pkg/transform/kubernetes/populators.go

```diff
@@ -161,3 +161,34 @@ func populateContainers(containers []spec.Container, cms []spec.ConfigMapMod, se
 	log.Debugf("containers after populating health: %s", string(b))
 	return cnts, nil
 }
+
+// Since we are automatically creating pvc from
+// root level persistent volume and entry in the container
+// volume mount, we also need to update the pod's volume field
+func populateVolumes(containers []api_v1.Container, volumeClaims []spec.VolumeClaim,
```

Comment:
>it just looks really weird to me if you have that much arguments that are always in the same struct

apart from the **look weird** part I don't think there is any other reason to not do it, also golang does not stop us from passing multiple arguments. We can always pass many arguments.

My only concern is that passing many things that a function does not need, makes that function to add unnecessary functionality because it has lot of information.